stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/FootnoteDecorator.kt

```diff
@@ -0,0 +1,197 @@
+package io.getstream.chat.android.ui.messages.adapter.viewholder.decorator
+
+import android.view.View
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isVisible
+import com.getstream.sdk.chat.adapter.MessageListItem
+import com.getstream.sdk.chat.utils.DateFormatter
+import com.getstream.sdk.chat.utils.extensions.isBottomPosition
+import com.getstream.sdk.chat.utils.extensions.isNotBottomPosition
+import com.getstream.sdk.chat.utils.extensions.updateConstraints
+import com.getstream.sdk.chat.utils.formatTime
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.client.utils.SyncStatus
+import io.getstream.chat.android.core.internal.exhaustive
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.messages.adapter.view.FootnoteView
+import io.getstream.chat.android.ui.messages.adapter.viewholder.GiphyViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.utils.extensions.getCreatedAtOrNull
+import io.getstream.chat.android.ui.utils.extensions.getUpdatedAtOrNull
+import io.getstream.chat.android.ui.utils.extensions.isEphemeral
+import io.getstream.chat.android.ui.utils.extensions.isGiphyNotEphemeral
+import io.getstream.chat.android.ui.utils.extensions.isInThread
+import io.getstream.chat.android.ui.utils.extensions.leftDrawable
+
+internal class FootnoteDecorator(
+    private val dateFormatter: DateFormatter,
+    private val isDirectMessage: Boolean,
+) : BaseDecorator() {
+
+    override fun decoratePlainTextWithFileAttachmentsMessage(
+        viewHolder: PlainTextWithFileAttachmentsViewHolder,
+        data: MessageListItem.MessageItem,
+    ) = setupFootnote(
+        viewHolder.binding.footnote,
```

Comment:
you could use `viewHolder.binding.run {}` and inline the `setupFootnote` call

Reply:
Matter of taste, but it will increase the amount of code-rows, I prefer to keep it so